src/Platform/Microsoft.Testing.Platform/TestHostOrcherstrator/TestHostOrchestratorManager.cs

```diff
@@ -48,4 +49,37 @@ public async Task<TestHostOrchestratorConfiguration> BuildAsync(ServiceProvider
 
         return new TestHostOrchestratorConfiguration([.. orchestrators]);
     }
+
+    public void AddTestHostOrchestratorApplicationLifetime(Func<IServiceProvider, ITestHostOrchestratorApplicationLifetime> testHostOrchestratorApplicationLifetimeFactory)
+    {
+        Guard.NotNull(testHostOrchestratorApplicationLifetimeFactory);
+        _testHostOrchestratorApplicationLifetimeFactories.Add(testHostOrchestratorApplicationLifetimeFactory);
+    }
+
+    internal async Task<ITestHostOrchestratorApplicationLifetime[]> BuildTestHostOrchestratorApplicationLifetimesAsync(ServiceProvider serviceProvider)
+    {
+        List<ITestHostOrchestratorApplicationLifetime> lifetimes = [];
+        foreach (Func<IServiceProvider, ITestHostOrchestratorApplicationLifetime> testHostOrchestratorApplicationLifetimeFactory in _testHostOrchestratorApplicationLifetimeFactories)
+        {
+            ITestHostOrchestratorApplicationLifetime service = testHostOrchestratorApplicationLifetimeFactory(serviceProvider);
+
+            // Check if we have already extensions of the same type with same id registered
+            if (lifetimes.Any(x => x.Uid == service.Uid))
+            {
+                ITestHostOrchestratorApplicationLifetime currentRegisteredExtension = lifetimes.Single(x => x.Uid == service.Uid);
+                throw new InvalidOperationException(string.Format(CultureInfo.InvariantCulture, PlatformResources.ExtensionWithSameUidAlreadyRegisteredErrorMessage, service.Uid, currentRegisteredExtension.GetType()));
+            }
```

Comment:
Why not simply `Where` to collect all potential duplicates (although we should not have any).

Reply:
I don't know. This is a copy of https://github.com/microsoft/testfx/blob/b3b81f44e18c4f1c5acdbb3974791b3023e158d1/src/Platform/Microsoft.Testing.Platform/TestHost/TestHostManager.cs#L104-L109 😄

Reply:
Oh right ok, not a big fan but let's use the same style then :)

Reply:
I'll create an issue to fix it in both places.